opal/runtime/opal_finalize.c

```diff
@@ -101,8 +113,9 @@ opal_finalize_util(void)
 
     opal_datatype_finalize();
 
-    /* finalize the class/object system */
-    opal_class_finalize();
+#if !(OPAL_HAVE_LD_FINI || OPAL_HAVE_ATTRIBUTE_DESTRUCTOR)
+    opal_cleanup ();
+#endif
```

Comment:
If we go with a NEED_LD_FINI "composite" AM_CONDITIONAL, above, then perhaps we should have a composite AC_DEFINE, too.  Perhaps "OPAL_NEED_MANUAL_DESTRUCTOR_INVOCATION"? (or perhaps something slightly less wordy ;-) )